src/plugins/fs/f2fs.c

```diff
@@ -325,28 +300,57 @@ BDFSF2FSInfo* bd_fs_f2fs_get_info (const gchar *device, GError **error) {
     if (!success) {
         /* error is already populated */
         bd_fs_f2fs_info_free (ret);
-        g_match_info_free (match_info);
-        g_regex_unref (regex);
         g_free (output);
         return NULL;
     }
 
-    item = g_match_info_fetch_named (match_info, "ssize");
-    ret->sector_size = g_ascii_strtoull (item, NULL, 0);
-    g_free (item);
+    lines = g_strsplit (output, "\n", 0);
+    g_free (output);
+    line_p = lines;
+
+    while (line_p && *line_p && !g_str_has_prefix (*line_p, "Info: sector size"))
+        line_p++;
+    if (!line_p || !(*line_p)) {
+        /* Sector size is not printed with dump.f2fs 1.15 */
+        ret->sector_size = 0;
+    } else {
+        /* extract data from something like this: "Info: sector size = 4096" */
+        val_start = strchr (*line_p, '=');
+        val_start++;
+        ret->sector_size = g_ascii_strtoull (val_start, NULL, 0);
+    }
 
-    item = g_match_info_fetch_named (match_info, "sectors");
-    ret->sector_count = g_ascii_strtoull (item, NULL, 0);
-    g_free (item);
+    line_p = lines;
+    while (line_p && *line_p && !g_str_has_prefix (*line_p, "Info: total FS sectors"))
+        line_p++;
+    if (!line_p || !(*line_p)) {
+        g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_PARSE, "Failed to parse NILFS2 file system information");
```

Comment:
NILFS2?

Reply:
Oops. You can guess from which file system I copied the code :-)